nucliadb_node/src/shards/providers/unbounded_cache/unbounded_writer.rs

```diff
@@ -96,13 +99,21 @@ impl ShardWriterProvider for UnboundedShardWriterCache {
         let mut cache = self.write();
         for entry in std::fs::read_dir(&self.shards_path)? {
             let entry = entry?;
-            let file_name = entry.file_name().to_str().unwrap().to_string();
+            let shard_id = entry.file_name().to_str().unwrap().to_string();
             let shard_path = entry.path();
-            match ShardWriter::open(file_name.clone(), &shard_path) {
+            let metadata = self.metadata_manager.get(shard_id.clone());
```

Comment:
Doing this is a problem if you serialise the metadata this way:
```rust
      let mut writer = BufWriter::new(File::create(metadata_path)?);
        serde_json::to_writer(
            &mut writer,
            &ShardMetadataFile {
                kbid: self.kbid.clone(),
                similarity: self.similarity,
                id: Some(self.id.clone()),
                channel: self.channel,
            },
        )?;
        writer.flush()?;
```
The problem is that you may read a partially written file, which can lead in undefined behaviour. I think the only way to this is by achieving atomicity in writing the metadata file.
A way of doing this is by using the `create and rename` strategy, basically, write the file a temp file and the rename it to the metadata file.